Analysis/Tasks/PWGHF/HFCandidateCreatorCascade.cxx

```diff
@@ -0,0 +1,304 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HFCandidateCreatorCascade.cxx
+/// \brief Reconstruction of heavy-flavour cascade decay candidates
+///
+
+#include "Framework/AnalysisTask.h"
+#include "DetectorsVertexing/DCAFitterN.h"
+#include "AnalysisDataModel/HFSecondaryVertex.h"
+#include "AnalysisCore/trackUtilities.h"
+#include "ReconstructionDataFormats/DCA.h"
+#include "ReconstructionDataFormats/V0.h"
+#include "AnalysisTasksUtils/UtilsDebugLcK0Sp.h"
+#include "AnalysisCore/HFSelectorCuts.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod::hf_cand_prong2;
+using namespace o2::analysis;
+
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+{
+  ConfigParamSpec optionDoMC{"doMC", VariantType::Bool, false, {"Perform MC matching."}};
+  workflowOptions.push_back(optionDoMC);
+}
+
+#include "Framework/runDataProcessing.h"
+
+//#define MY_DEBUG
+
+#ifdef MY_DEBUG
+using MyTracks = aod::BigTracksMC;
+#define MY_DEBUG_MSG(condition, cmd) \
+  if (condition) {                   \
+    cmd;                             \
+  }
+#else
+using MyTracks = aod::BigTracks;
+#define MY_DEBUG_MSG(condition, cmd)
+#endif
+
+/// Reconstruction of heavy-flavour cascade decay candidates
+struct HFCandidateCreatorCascade {
+
+  Produces<aod::HfCandCascBase> rowCandidateBase;
+
+  Configurable<double> d_bZ{"d_bZ", 5., "magnetic field"};
+  Configurable<bool> b_propDCA{"b_propDCA", true, "create tracks version propagated to PCA"};
+  Configurable<double> d_maxR{"d_maxR", 200., "reject PCA's above this radius"};
+  Configurable<double> d_maxDZIni{"d_maxDZIni", 4., "reject (if>0) PCA candidate if tracks DZ exceeds threshold"};
+  Configurable<double> d_minParamChange{"d_minParamChange", 1.e-3, "stop iterations if largest change of any X is smaller than this"};
+  Configurable<double> d_minRelChi2Change{"d_minRelChi2Change", 0.9, "stop iterations is chi2/chi2old > this"};
+  Configurable<bool> b_doValPlots{"b_doValPlots", true, "do validation plots"};
+
+  // for debugging
+#ifdef MY_DEBUG
+  Configurable<std::vector<int>> v_labelK0Spos{"v_labelK0Spos", {729, 2866, 4754, 5457, 6891, 7824, 9243, 9810}, "labels of K0S positive daughters, for debug"};
+  Configurable<std::vector<int>> v_labelK0Sneg{"v_labelK0Sneg", {730, 2867, 4755, 5458, 6892, 7825, 9244, 9811}, "labels of K0S negative daughters, for debug"};
+  Configurable<std::vector<int>> v_labelProton{"v_labelProton", {717, 2810, 4393, 5442, 6769, 7793, 9002, 9789}, "labels of protons, for debug"};
+#endif
```

Comment:
Underscores are not allowed. Please remove them together with the prefixes everywhere in the new code.

Reply:
done